bip-txrelayv2.mediawiki

```diff
@@ -0,0 +1,49 @@
+<pre>
+  BIP: XXX
+  Layer: Peer Services
+  Title: Transaction Relay V2
+  Author: Antoine Riard <btc@ariard.me>
+  Comments-Summary: No comments yet.
+  Comments-URI: https://github.com/bitcoin/bips/wiki/Comments:BIP-0339
+  Status: Draft
+  Type: Standards Track
+  Created: 2024-09-05
+  License: GPLv2
+</pre>
+
+==Abstract==
+
+This BIP proposes a new Bitcoin transaction relay protocol by introducing new
+signaling support, which can be further used by nodes to negotiate new transaction
+relay mechanisms and policies.
+
+==Motivation==
+
+Historically, nodes have been exchanging transactions on the Bitcoin peer-to-peer
+network by sending an inv, and if the transaction has not been discovered and processed
+yet by the other peer, sending a dedicated tx message.
+
+Sending an unnannounced tx message has always been considered in conformity with
+the protocol, however this behavior creates a denanonymization vector if leveraged
+by mass connectors.
+
+We can eliminate this concern by using node service bits to signal support for
+a new v2 transaction relay protocol. Within this protocol, nodes can negotiate new
+transaction relay mechanisms and policies.
+
+==Specification==
+
+Peers supporting the v2 transaction relay protocol signal support by adverstising
+the 13th bit service flag in the addr p2p messages (`ADDR` and `ADDRV2`).
```

Comment:
The Specification seems to define how to advertise a new network service, but it should perhaps elaborate what this service entails.

Reply:
No, it’s intentional. The mechanisms and policies supported should be described in a distinct BIP, e.g if some class of messages are disabled like it was proposed by BIP338, which is a distinct mechanism than behavior regarding unsolicited transactions.